pycls/core/checkpoint.py

```diff
@@ -57,9 +57,7 @@ def has_checkpoint():
 
 def save_checkpoint(model, model_ema, optimizer, epoch, test_err, ema_err):
     """Saves a checkpoint and also the best weights so far in a best checkpoint."""
-    # Save checkpoints only from the main process
```

Comment:
Is there a reason to move this code to line 76?

Reply:
Yes, we need all processes to call state_dict() since there is a synchronization of processes in FSDP's implementation.